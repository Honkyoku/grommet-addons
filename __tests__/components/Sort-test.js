// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Sort from '../../src/js/components/Sort';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Sort', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Sort options={[
        { label: 'First', value: 'first' },
        { label: 'Second', value: 'second' }
      ]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
