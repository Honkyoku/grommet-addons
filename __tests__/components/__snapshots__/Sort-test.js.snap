exports[`Sort has correct default options 1`] = `
<div
  className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--align-center grommetux-box--pad-none"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    className="grommetux-select"
    onClick={[Function bound _onAddDrop]}>
    <input
      className="grommetux-input grommetux-select__input"
      disabled={true}
      id={undefined}
      name={undefined}
      value={undefined} />
    <button
      aria-label="Select Icon"
      className="grommetux-button grommetux-button--plain grommetux-button--icon grommetux-select__control"
      disabled={false}
      href={undefined}
      id={undefined}
      onBlur={[Function onBlur]}
      onClick={[Function bound _onAddDrop]}
      onFocus={[Function onFocus]}
      onMouseDown={[Function onMouseDown]}
      onMouseUp={[Function onMouseUp]}
      type="button">
      <span
        className="grommetux-button__icon">
        <svg
          aria-label="caret-down"
          className="grommetux-control-icon grommetux-control-icon-caret-down grommetux-control-icon--responsive"
          height="24px"
          role="img"
          version="1.1"
          viewBox="0 0 24 24"
          width="24px">
          <polygon
            fill="none"
            points="22 8 12 20 2 8"
            stroke="#000000"
            strokeWidth="2" />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-none grommetux-box--flex-off"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <button
      aria-label={undefined}
      className="grommetux-button grommetux-button--plain grommetux-button--icon"
      disabled={false}
      href={undefined}
      id={undefined}
      onBlur={[Function onBlur]}
      onClick={[Function bound bound _onChangeDirection]}
      onFocus={[Function onFocus]}
      onMouseDown={[Function onMouseDown]}
      onMouseUp={[Function onMouseUp]}
      type="button">
      <span
        className="grommetux-button__icon">
        <svg
          aria-label="link-down"
          className="grommetux-control-icon grommetux-control-icon-link-down grommetux-control-icon--responsive"
          height="24px"
          role="img"
          version="1.1"
          viewBox="0 0 24 24"
          width="24px">
          <path
            d="M12,2 L12,22 M3,11 L12,2 L21,11"
            fill="none"
            stroke="#000000"
            strokeWidth="2"
            transform="matrix(1 0 0 -1 0 24)" />
        </svg>
      </span>
    </button>
    <button
      aria-label={undefined}
      className="grommetux-button grommetux-button--plain grommetux-button--icon"
      disabled={false}
      href={undefined}
      id={undefined}
      onBlur={[Function onBlur]}
      onClick={[Function bound bound _onChangeDirection]}
      onFocus={[Function onFocus]}
      onMouseDown={[Function onMouseDown]}
      onMouseUp={[Function onMouseUp]}
      type="button">
      <span
        className="grommetux-button__icon">
        <svg
          aria-label="link-up"
          className="grommetux-control-icon grommetux-control-icon-link-up grommetux-control-icon--responsive"
          height="24px"
          role="img"
          version="1.1"
          viewBox="0 0 24 24"
          width="24px">
          <path
            d="M12,2 L12,22 M3,11 L12,2 L21,11"
            fill="none"
            stroke="#000000"
            strokeWidth="2" />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
